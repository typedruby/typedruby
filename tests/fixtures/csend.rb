# @typedruby
def test_csend(~String x) => nil
  x&.length
  "hi"&.length
  x&.length + 1
  nil
end
