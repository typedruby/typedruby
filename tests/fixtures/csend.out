error: conditional-send on a non-nilable receiver

        @ __ROOT__/tests/fixtures/csend.rb:4
      4 |    "hi"&.length
             ^^^^ receiver
        @ __ROOT__/tests/fixtures/csend.rb:4
      4 |    "hi"&.length
             ^^^^^^^^^^^^ in this invocation

error: Union member NilClass does not respond to #+

        @ __ROOT__/tests/fixtures/csend.rb:5
      5 |    x&.length + 1
             ^^^^^^^^^ Integer | NilClass
